certbot-nginx/certbot_nginx/nginxparser.py

```diff
@@ -248,7 +248,7 @@ def is_dirty(self):
         """Recurse through the parse tree to figure out if any sublists are dirty"""
         if self.dirty:
             return True
-        return any((isinstance(x, list) and x.is_dirty() for x in self))
+        return any((isinstance(x, list) and x.is_dirty() for x in self)) # type: ignore
```

Comment:
Changing the code here to `...isinstance(x, UnspacedList)...` and removing the `type: ignore` seems to work. This should be safe because if `x` is really a `list` and not `UnspacedList`, it won't have the `is_dirty` attribute and blow up.

Reply:
Hrm, I *thought* that I had tried that, but yes, that *does* seem to work. Thanks!